code/iaas/storage-service/src/main/java/io/cattle/platform/storage/api/filter/ExternalTemplateInstanceFilter.java

```diff
@@ -55,14 +60,34 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
         return super.create(type, request, next);
     }
 
-    protected Image validateImageUuid(SchemaFactory schemaFactory, String uuid, Long registryCredentialId) {
+    protected Image validateImageUuid(SchemaFactory schemaFactory, String uuid, Instance instance) {
         try {
             Image image = storageService.registerRemoteImage(uuid);
             if (image == null) {
                 return null;
             }
-            if (registryCredentialId != null) {
-                objectManager.setFields(image, IMAGE.REGISTRY_CREDENTIAL_ID, registryCredentialId);
+            Long id = instance.getRegistryCredentialId();
+            if (id == null && image.getFormat().equalsIgnoreCase(DockerStoragePoolConstants.DOCKER_FORMAT)) {
+                String type = schemaFactory.getSchemaName(StoragePool.class);
+                List<?> storagePools = locator.getResourceManagerByType(type).list(type, new HashMap<>(), new ListOptions());
```

Comment:
@son ResourceManager does authorization checks based on the API logic.  ObjectManager is more raw database access.  When in the API tier and you need to look up objects it's safe to use ResourceManager because of the authorization checks.  Imagine that one day we have more complex authorization like IAM.

Also, ResourceManager mimics API requests of GET, POST, PUT, etc.  So by calling `list()` you are getting the equivalent of `/v1/storagepools` which is already scoped to objects the user is allow to access.